### ISSUE TITLE
Allow `"..."` to be used in `resolve.extensions` to extend the default values

### ISSUE THREAD

**Issue by glen-84** [webpack/webpack#13023]
## Feature request

Allow `"..."` to be used in `resolve.extensions` to extend the default values.

**What is the expected behavior?**

```js
resolve: {
    extensions: ["...", ".ts", ".vue"] // Allow this.
}
```

**What is motivation or use case for adding/changing the behavior?**

Extend the defaults without having to redefine them.

**How should this be implemented in your opinion?**

Similar to how it's implemented in other places, I assume.

**Are you willing to work on this yourself?**

Not in the short term.

**Comment by sokra** [comment]
Yes that should work this way. Doesn't it work for you? Could you create a repro for your problem?

Note that the order might matter any you actually want `extensions: [".ts", ".vue", "..."]`

**Comment by glen-84** [comment]
It doesn't work, at the beginning or at the end.

Does the code exist for supporting this? If it does, I can create a repro, but it seems like a waste of time for such a small thing. 🙂

**Comment by glen-84** [comment]
`5.28.0`

**Comment by alexander-akait** [comment]
Please create example, works fine for me

**Comment by glen-84** [comment]
I think that this is a documentation bug. The documentation [states](https://webpack.js.org/configuration/resolve/#resolveextensions) that `.mjs` is included, but it [seems](https://github.com/webpack/webpack/blob/f3dce69e9df5f778fb4850662b0f16f9fd9951d2/lib/config/defaults.js#L1000) that it's not?

**Comment by alexander-akait** [comment]
@glen-84 Yep, ideally you don't need this option for JS `import`, because using `import` requires `extension` at the end (i.e. `import foo from './something.mjs'`, can be `js` extension too, so webpack don't need apply `resolve.extensions`), when you write `import { foo } from 'package';` we use `exports`/`main`/`module`/etc fields from `package.json`. 

Also I strongly recommend always use extensions for any `import`
